src/main/java/io/asyncer/r2dbc/mysql/MySqlConnection.java

```diff
@@ -303,13 +303,27 @@ public IsolationLevel getTransactionIsolationLevel() {
         return currentLevel;
     }
 
+    /**
+     * Gets session transaction isolation level(Only for testing).
+     *
+     * @return session transaction isolation level.
+     */
+    IsolationLevel getSessionTransactionIsolationLevel() {
+        return sessionLevel;
+    }
+
     @Override
     public Mono<Void> setTransactionIsolationLevel(IsolationLevel isolationLevel) {
         requireNonNull(isolationLevel, "isolationLevel must not be null");
 
-        // Set next transaction isolation level.
-        return QueryFlow.executeVoid(client, "SET TRANSACTION ISOLATION LEVEL " + isolationLevel.asSql())
-            .doOnSuccess(ignored -> setIsolationLevel(isolationLevel));
+        // Set subsequent transaction isolation level.
+        return QueryFlow.executeVoid(client, "SET SESSION TRANSACTION ISOLATION LEVEL " + isolationLevel.asSql())
+            .doOnSuccess(ignored -> {
+                this.sessionLevel = isolationLevel;
+                if (!this.isInTransaction()) {
```

Comment:
I'm not sure if the condtion is correct to set current level even if we reset isolation level when committing or rolling back.

Reply:
@mirromutth, Could you please take a look?

It seems accurate since that `sessionLevel` denotes the isolation level for upcoming and subsequent transactions. 
When in a transaction, `currentLevel` indicates the isolation level of the current transaction. when not in a transaction, `currentLevel` represents the isolation level for the very next transaction.

Reply:
Maybe I should populate the current isolation level at the begining of beginTransaction().

Reply:
I believe the current implementation effectively deals with that scenario:

Case 1: When the isolation level is set within a transaction, the currentLevel will equal the sessionLevel following a rollback or commit. (Note: r2dbc-mysql doesn't allow a double begin.)
Case 2: If the isolation level is set outside of a transaction, this method will adjust both the session level and the current level.

As a result, in both scenarios, when you are not in a transaction(or before executing `beginTransaction()`), the condition currentLevel == sessionLevel is maintained.